### DISCUSSION TITLE
switch for AnimationNodeTransition changed from int to string !? why?

### DISCUSSION THREAD

**Discussion by dr3muchiy** [godotengine/godot-proposals#6249] [Scripting]
I was very happy when saw AnimationNodeTransition can now have more than 32 Inputs.

But when I see switch  -   "transition_request"  String  type -  I was very upset.

Old version AnimationNodeTransition switch by "current"  Int  type. 
It was good - just write  Inputs array cell number there - and everything is in order - there is no senseless performance loss here.

But if switch have String type,  godot in a loop will iterate over the cells and check transition_request==Inputs[i].name to find the int cell number

why performance is thrown down the drain? When possible don't force godot  iterate Inputs[], but simply set cell through int variable.

In my project will be long AnimationNodeTransition.(they are not yet completely filled, but will be) Switching in Transition can happen quite often. And if there are many characters on the map.  these iterations in the loop can eat a significant share of performance.
![image](https://user-images.githubusercontent.com/122445959/217275591-dd437992-39ca-477f-80d6-94c4df17f4f8.png)

**Comment by QbieShay** [comment]
Did you observe a decrease in performance?

**Reply by dr3muchiy** [reply]
> Did you observe a decrease in performance?

At this stage of development - not . So far, I have few characters in the location. Probably not be noticeable even at 100 characters . But here, meaningless operations are performed. If such decisions are repeated in another place, in the third place, as a result, everything together will result in noticeable losses.
